lib/sprite_animation.dart

```diff
@@ -4,6 +4,81 @@ import 'dart:ui';
 import 'extensions/vector2.dart';
 import 'sprite.dart';
 
+class SpriteAnimationFrameData {
+  /// Coordinates of the sprite of this Frame
+  final Vector2 srcPosition;
+
+  /// Size of the sprite of this Frame
+  final Vector2 srcSize;
+
+  /// The duration to display it, in seconds.
+  double stepTime;
+
+  SpriteAnimationFrameData({
+    @required this.srcPosition,
+    @required this.srcSize,
+    @required this.stepTime,
+  });
+}
+
+class SpriteAnimationData {
+  List<SpriteAnimationFrameData> frames;
+  final bool loop;
+
+  /// Creates a SpriteAnimationData from the given [frames] and [loop] parameters
+  SpriteAnimationData(this.frames, {this.loop = true});
+
+  /// Takes some parameters and automatically calculate and create the frames for the sprite animation data
+  ///
+  /// [amount] The total amount of frames present on the image
+  /// [stepTimes] A list of times (in seconds) of each frame, should have a length equals to the amount parameter
+  /// [textureSize] The size of each frame
+  /// [amountPerRow] An optional parameter to inform how many frames there are on which row, useful for sprite sheets where the frames as disposed on multiple lines
+  /// [texturePosition] An optional parameter with the initial coordinate where the frames begin on the image, default to (top: 0, left: 0)
+  /// [loop] An optional parameter to inform if this animation loops or has a single iteration, defaults to true
+  SpriteAnimationData.variable({
+    @required int amount,
+    @required List<double> stepTimes,
+    @required Vector2 textureSize,
+    int amountPerRow,
+    Vector2 texturePosition,
+    this.loop = true,
+  })  : assert(amountPerRow == null || amount >= amountPerRow),
+        assert(stepTimes != null) {
+    amountPerRow ??= amount;
+    texturePosition ??= Vector2.zero();
+    frames = List<SpriteAnimationFrameData>(amount);
+    for (int i = 0; i < amount; i++) {
+      final position = Vector2(
+        texturePosition.x + (i % amountPerRow) * textureSize.x,
+        texturePosition.y + (i ~/ amountPerRow) * textureSize.y,
+      );
```

Comment:
you can extract `texturePosition` outside like `texturePosition + Vector2(\n...`

Reply:
Not sure about that, I think I prefer the current implementation.